### ISSUE TITLE
lake exe export_equations doesn't build

### ISSUE THREAD

**Issue by vlad902** [teorth/equational_theories#501]
I get the following build error on `lake exe export_equations` on main:
```
error: ././././scripts/export_equations.lean:27:39: application type mismatch
  Json.arr fun h ↦ Subtype.map (fun x ↦ ?m.281 x) h laws
argument
  fun h ↦ Subtype.map (fun x ↦ ?m.281 x) h laws
has type
  (∀ (a : RBNode Name fun x ↦ Unit), RBNode.WellFormed Name.quickCmp a → ?m.273 ((fun x ↦ ?m.281 x) a)) →
    Subtype ?m.273 : Sort (max 1 ?u.261)
but is expected to have type
  Array Json : Type
error: ././././scripts/export_equations.lean:27:52: invalid constructor ⟨...⟩, expected type must be an inductive type with only one constructor
  RBNode Name fun x ↦ Unit
error: Lean exited with code 1
Some required builds logged failures:
- scripts.export_equations
error: build failed
```

_Originally posted by @vlad902 in https://github.com/teorth/equational_theories/issues/142#issuecomment-2405393119_

**Comment by vlad902** [comment]
And while we're here, is it possible to have this write to stdout instead of a fixed file name? It's not a given that it will be invoked from the root dir of the project repo, and it better matches the output of `export_implications`

**Comment by amirlb** [comment]
claim

**Comment by amirlb** [comment]
I ran out of idea trying to get this script to work. It's simple to fix the types, and to change inputs and outputs. However

* `magmaLawExt.getState env` returns an empty list, and anyway I don't see a way to make it work as it returns only `NameSet` and not actual values.
* `getMagmaLaws` which does it more properly (using `evalConstCheck`) crashes with `uncaught exception: unknown declaration`. This should never happen since the "check" part of the function name checks that the binding exists in the environment before evaluating. I tried several variations on this theme and couldn't find a way to get the values representing the laws.

In short, I'm out of my depth. Leaving this for somebody else to figure out.

**Comment by amirlb** [comment]
disclaim

**Comment by amirlb** [comment]
Looks like @0art0 fixed it a few hours ago.

I still have a branch with
* Output to stdout
* Adding a string form of the laws to the output `0 * 1 = 1 * 0` in addition to `{"lhs": {"left": {"leaf": 0}, "right": {"leaf": 1}}, "rhs": {"left": {"leaf": 1}, "right": {"leaf": 0}}}`
* Specifying a module to fetch the laws from
Do we want that?

**Comment by vlad902** [comment]
The first two sound great, the third sounds good too but it should probably be like `extract_implications` where it defaults to 'all' if none is given.

**Comment by amirlb** [comment]
Created a PR at https://github.com/teorth/equational_theories/pull/561

I'm not sure what the proper procedure is for this issue, hope a maintainer catches and merges it...

**Comment by pitmonticone** [comment]
Thanks @vlad902 for writing the issue and @amirlb for opening #561. 

Would #561 close this issue (#501)?

**Comment by vlad902** [comment]
Terence added the `project-task` label, so you can claim it now and put it on the project board.

**Comment by pitmonticone** [comment]
Yes, @amirlb feel free to claim and propose the PR.

**Comment by amirlb** [comment]
Looks like this was already merged!

So we can definitely close the issue now. Integrating this script into visualizations and other tools is #482
